### PR TITLE
Update the calls to axes to evolve_info

### LINKED ISSUE #72: Replace `Grid::axes` with `Grid::evolve_info`
`Grid::axes` is too restrictive, see the discussion here: <https://github.com/NNPDF/pineappl/issues/197>. Therefore it should be replaced with `Grid::evolve_info`.

### PR DESCRIPTION
**Pull request by scarlehoff**
Closes #72 

If I understood `evolve_info` correctly this should be doing the same as was done before. The only thing that I'm not sure about is what is the implicit masking that `axes` was doing? Or maybe `axes` assumed that all orders had the exact same x/q structure?

### CODE REVIEW

File: src/pineko/evolve.py
Comment (scarlehoff) [inline src/pineko/evolve.py:105]:
```suggestion
    # Add a +1 to the orders for the difference in convention between nnpdf and pineappl
    is_fns = int(check.is_fonll_b(tcard["FNS"], pineappl_grid.lumi()))
    max_as = 1 + tcard["PTO"] + is_fns
    max_al = 1 + tcard["QED"]
```

**Comment by cschwan** [comment]
@scarlehoff the old behaviour is to use all orders (mask set to all true), which potentially makes the EKOs larger than needed. That's not wrong just less performant.

File: src/pineko/evolve.py
Comment (andreab1997) [inline src/pineko/evolve.py:153]: Why did you remove this?

**Review by andreab1997 — APPROVED** [review]

File: src/pineko/evolve.py
Comment (scarlehoff) [reply src/pineko/evolve.py:153]: afaics it's not used anywhere in that function

**Review by AleCandido — CHANGES_REQUESTED** [review]
Do not ask for a maximum value, just create a trivial mask

File: src/pineko/check.py
Comment (AleCandido) [reply src/pineko/check.py:66]: https://github.com/NNPDF/pineko/blob/2cbedfb06f19413cc18d5b86c41170287267a1da/src/pineko/evolve.py#L25

It will be soon as a default inside PineAPPLpy

Comment (AleCandido) [reply src/pineko/check.py:66]: Driooed from where?

Comment (andreab1997) [reply src/pineko/check.py:66]: Yes, and that is more or less what I did in #75. And the reason for that was exactly to avoid that this check depended on max_as and max_al

Comment (AleCandido) [reply src/pineko/check.py:66]: So, if you intend "dropped from the grid", of course the trivial mask would be adaptive, since it depends on `grid.orders()`, so there is nothing that `max_as/l` could cut.

Of course, HO can't be dropped from the EKO, since it is a resummed object and they are not stored separately.

Comment (scarlehoff) [reply src/pineko/check.py:66]: Yes, this concerns only the grid.

**Review by AleCandido — APPROVED** [review]
Improvements (i.e. simplifications) postponed to #85. The rest looks perfect to me :)

**Comment by scarlehoff** [comment]
In order to test this I'm redoing all fktables in theory 400.

DIS agree perfectly with what @andreab1997 has in dom for theories 4054 (NLO) and 4244 (NNLO).
ATLAS FKTables have already finished and they also agree*. CMS tables are under construction.

However, I've found a bug for `ATLASDY2D8TEV`. I don't think it is introduced by this PR but rather a bug in `evolve_info` (or in the combination of `evolve_info`, `evolve` and grids converted from applgrid) so I would wait until this is understood to merge this PR. I've written about the bug here: https://github.com/NNPDF/pineko/issues/72#issuecomment-1484555199

*btw, a bit more than 24 hours to finish all ATLAS FkTable -no jets- using a single core in the eko template.

**Comment by andreab1997** [comment]
> In order to test this I'm redoing all fktables in theory 400.
> 
> DIS agree perfectly with what @andreab1997 has in dom for theories 4054 (NLO) and 4244 (NNLO). ATLAS FKTables have already finished and they also agree*. CMS tables are under construction.
> 
> However, I've found a bug for `ATLASDY2D8TEV`. I don't think it is introduced by this PR but rather a bug in `evolve_info` (or in the combination of `evolve_info`, `evolve` and grids converted from applgrid) so I would wait until this is understood to merge this PR. I've written about the bug here: [#72 (comment)](https://github.com/NNPDF/pineko/issues/72#issuecomment-1484555199)
> 
> *btw, a bit more than 24 hours to finish all ATLAS FkTable -no jets- using a single core in the eko template.

What is the status of this? Can this be merged?

**Comment by scarlehoff** [comment]
Ok, this is ready. I've updated `pyproject.toml` since this needs pineappl > 0.6.0 alpha 5

**Comment by AleCandido** [comment]
@scarlehoff can you run `poetry lock` and upload the updated lock file?

**Comment by scarlehoff** [comment]
Yes... but poetry seems to be having a really hard time resolving the dependencies... (it's about to do 10 minutes...) (I had installed pineappl from the repo so forgot about the lockfile)

**Comment by AleCandido** [comment]
Are you using Poetry 1.4.1? They recently improved a lot performances (recently = months).

**Comment by AleCandido** [comment]
```
❯ poetry lock
Updating dependencies
Resolving dependencies... (7.1s)

Writing lock file
```

**Comment by AleCandido** [comment]
@scarlehoff I'd be in favor of merging :)

**Comment by scarlehoff** [comment]
> ❯ poetry lock
> Updating dependencies
> Resolving dependencies... (7.1s)
> 
> Writing lock file

Thanks. I'm at 2490s now... And no, I have poetry 1.2.2. Time to update I guess.

**Pull request merged**
